### ISSUE TITLE
light curve peak normalization

### ISSUE THREAD

**Issue by joaofrancafisica** [sncosmo/sncosmo#273]
Hi there,

I have a few questions about SNcosmo multi-band light curve fit, especially using the salt2 model.

It seems to me that the code realizes a different normalization for each band in data (or in the theoretical flux given, in this case, by the salt2 x0*(M0+x1M1)exp(cC))). I would like to know if this normalization uses only the zeropoint or synthetic data too.

I said that because I tried to reproduce a simple light curve fit with the same data and same model and I'm having this trouble with normalization. I'm sending the image of both fit's.

![fit](https://user-images.githubusercontent.com/51277157/84100363-15b35a00-a9e2-11ea-8735-70f27c1034f5.png)
![sncosmofit](https://user-images.githubusercontent.com/51277157/84100364-16e48700-a9e2-11ea-9d42-0a5049ed581f.png)

Thanks in advance

**Comment by kbarbary** [comment]
The SALT2 model does have some flexibility to fit the peak in different bands via the color parameter `c`. A larger `c` value will increase the peak in `sdssi` relative to the peak in `sdssg` for example. However, it can't match an arbitrary number of bands peaks. You can see in the second fit that you posted that it underestimates the peak in `sdssi` and overestimates the peak in `sdssr`. This is because increasing `c` would make the fit better in `sdssi` but worse in `sdssr`, so this is (probably) the optimal `c` value.

I'm not sure if that answers your question. It might help to explain what code/model you used to produce the first fit result? By eye, it looks like increasing `c` would lead to a much better fit there, so I suspect `c` has been fixed to a too-low value there.

**Comment by joaofrancafisica** [comment]
Thanks for your reply!

Initially, I thought that it would be the c parameter too. But I fixed the parameter c = 0 in sncosmo and the result wasn't so bad as the first one. Do you believe it is an optimization problem in this case (my code)? I'm sending the sncosmo fit with c=0 in attachment.
![salt2fitc0](https://user-images.githubusercontent.com/51277157/84190977-2dd0bb00-aa6e-11ea-998c-c740d5d73d3d.png)

**Comment by kbarbary** [comment]
>  Do you believe it is an optimization problem in this case (my code)?

It's hard to say without knowing more about what your code is doing or what model implementation it is using.

Also I noticed that the two early "rising" light curve points are not visible in your plots, but are visible in the sncosmo plots. I doubt that is the cause of the discrepancy though.

**Comment by joaofrancafisica** [comment]
I see.

One last question: In the internal files in sncosmo (I'm attaching an image), you are normalizing the theoretical flux by zero-point, right? Is this zeropoint the one in data file? 

![git](https://user-images.githubusercontent.com/51277157/84210479-31c10500-aa8f-11ea-950e-2f3a300d02a7.png)

Thanks for all!

**Comment by kbarbary** [comment]
> In the internal files in sncosmo (I'm attaching an image), you are normalizing the theoretical flux by zero-point, right? Is this zeropoint the one in data file?

Yep. The model flux is scaled to match the zero point of the data, so that the model flux and observed flux can be directly compared.

Please ask more questions if you are still seeing discrepancies you'd like help tracking down!
